### PR TITLE
RBAC v2 DELETE /groups/{group_id}/hosts/{host_id_list} 

### PR DESCRIPTION
**Pull request by mjholder**
# Overview

This PR is being created to address [RHINENG-15846](https://issues.redhat.com/browse/RHINENG-15846).
(A description of your PR's changes, along with why/context to the PR, goes here.)

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: lib/middleware.py [hunk 1]
```diff
@@ -302,16 +302,16 @@ def post_rbac_workspace(name, parent_id, description) -> UUID | None:
     return UUID(resp_data["id"])
 
 
-def rbac_create_ungrouped_hosts_workspace(identity: Identity) -> UUID | None:  # noqa: ARG001, used later
+def rbac_create_ungrouped_hosts_workspace() -> UUID | None:  # noqa: ARG001, used later
```
Comment (kruai) [inline lib/middleware.py:305]: I had assumed that we would need the Identity for this call, but I was wrong - we just pass the header through. Since we don't actually need it, we can remove the noqa.
```suggestion
def rbac_create_ungrouped_hosts_workspace() -> UUID | None: 
```

**Review by kruai — APPROVED** [review]

**Review by FabriciaDinizRH — DISMISSED** [review]
LGTM!

**Review by FabriciaDinizRH — DISMISSED** [review]

**Review by kruai — APPROVED** [review]
